jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/HandlerRegistry.java

```diff
@@ -12,44 +12,70 @@
 import io.opentelemetry.contrib.jfr.metrics.internal.cpu.ContextSwitchRateHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.cpu.LongLockHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.cpu.OverallCPULoadHandler;
-import io.opentelemetry.contrib.jfr.metrics.internal.memory.G1GarbageCollectionHandler;
+import io.opentelemetry.contrib.jfr.metrics.internal.memory.G1HeapSummaryHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.memory.GCHeapSummaryHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.memory.ObjectAllocationInNewTLABHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.memory.ObjectAllocationOutsideTLABHandler;
+import io.opentelemetry.contrib.jfr.metrics.internal.memory.ParallelHeapSummaryHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.network.NetworkReadHandler;
 import io.opentelemetry.contrib.jfr.metrics.internal.network.NetworkWriteHandler;
+import java.lang.management.ManagementFactory;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
 
 final class HandlerRegistry {
   private static final String INSTRUMENTATION_NAME = "io.opentelemetry.contrib.jfr";
   private static final String INSTRUMENTATION_VERSION = "1.7.0-SNAPSHOT";
 
   private final List<RecordedEventHandler> mappers;
 
+  private static final Map<String, List<Supplier<RecordedEventHandler>>> HANDLERS_PER_GC =
+      Map.of(
+          "G1",
+          List.of(G1HeapSummaryHandler::new),
+          "Parallel",
+          List.of(ParallelHeapSummaryHandler::new));
+
   private HandlerRegistry(List<? extends RecordedEventHandler> mappers) {
     this.mappers = new ArrayList<>(mappers);
   }
 
   static HandlerRegistry createDefault(MeterProvider meterProvider) {
-    var meter =
-        meterProvider
-            .meterBuilder(INSTRUMENTATION_NAME)
-            .setInstrumentationVersion(INSTRUMENTATION_VERSION)
-            .build();
+    var handlers = new ArrayList<RecordedEventHandler>();
+    var seen = new HashSet<String>();
+    for (var bean : ManagementFactory.getGarbageCollectorMXBeans()) {
+      var name = bean.getName();
+      for (var gcType : HANDLERS_PER_GC.keySet()) {
+        if (name.contains(gcType)
+            && !seen.contains(gcType)
+            && HANDLERS_PER_GC.get(gcType) != null) {
```

Comment:
This check is redundant since we're iterating through the keyset of of `HANDLERS_PER_GC`. I suppose it could happen if someone set a map entry to null, but that's a pretty unlikely error.

Reply:
I know - but Errorprone insists on the null check.